hydrabus_framework/core/engine.py

```diff
@@ -0,0 +1,122 @@
+
+            # Prompt.
+            'username': '#3399ff',
+            'pound': '#3399ff',
+            'path': '#ff0000 bold',
+        })
+        self.prompt = [
+            ('class:username', '[hbf] '),
+            ('class:path', ''),
+            ('class:pound', '> '),
+        ]
+        self.command = [
+            {"name": "show", "descr": "modules|options: Displays modules list, or module options", "run": show},
+            {"name": "help", "descr": "Help menu", "run": hbf_help},
+            {"name": "exit", "descr": "Exit the console", "run": hbf_exit},
+            {"name": "use", "descr": "Load a module by name", "run": use},
+            {"name": "run", "descr": "Run the selected module", "run": run_module},
+            {"name": "back", "descr": "Move back from the current context", "run": back},
+            {"name": "set", "descr": "Sets a context-specific variable to a value", "run": set_options}
+        ]
+
+    def update_prompt(self, module_name):
+        """
+        Update the user prompt
+        :param module_name: The current module name
+        """
+        if module_name != "":
+            self.prompt = [
+                ('class:username', '[hbf] '),
+                ('class:path', '({})'.format(module_name)),
+                ('class:pound', '> '),
+            ]
+        else:
+            self.prompt = [
+                ('class:username', '[hbf] '),
+                ('class:path', ''),
+                ('class:pound', '> '),
+            ]
+
+    def _list_modules(self):
+        """
+        Generate modules path and attributes list
+        :return: List of available modules
+        """
+        modules = []
+        module_name = "hbfmodules"
+
+        try:
+            package = import_module(module_name)
+        except ImportError:
+            self.logger.print('Package {} not found...'.format(module_name), "error")
+            sys.exit(1)
+        for loader, module, is_pkg in pkgutil.walk_packages(package.__path__, prefix=package.__name__ + '.'):
+            imported_module = import_module(module)
+            for x in dir(imported_module):
+                obj = getattr(imported_module, x)
+                if inspect.isclass(obj) and issubclass(obj, ABaseModule) and obj is not ABaseModule:
+                    module_path = module.replace('hbfmodules.', '').replace('.', '/')
+                    modules.append({"path": module_path, "class": obj})
+        return modules
+
+    def handler(self, command):
+        """
+        User console command handler
+        :param command: User input
+        """
+        for cmd in self.command:
+            if command.split(" ")[0] == cmd["name"]:
+                if len(inspect.getfullargspec(cmd["run"])[0]) == 2:
+                    cmd["run"](self, command)
+                elif len(inspect.getfullargspec(cmd["run"])[0]) == 1:
+                    cmd["run"](self)
+                else:
+                    cmd["run"]()
+                break
```

Comment:
You should use varargs in your framework commands to simplify the dispatching and increase the code flexibility

Reply:
Done [#01a8ef3](https://github.com/hydrabus-framework/framework/commit/01a8ef3a096bb762626bc71cba7fa939c6f3ffba)